tests/unit/script/textinput.rs

```diff
@@ -600,6 +600,13 @@ fn test_textinput_set_selection_with_direction() {
     assert!(textinput.selection_origin.is_some());
     assert_eq!(textinput.selection_origin.unwrap().line, 0);
     assert_eq!(textinput.selection_origin.unwrap().index, 6);
+
+    textinput = text_input(Lines::Multiple, "\n\n");
```

Comment:
Let's test also with only `\n`, which is the case we're concerned about. Also, let's add a crashtest for this based on @mateon1's test-case, into `tests/wpt/web-platform-tests/selection/`. See https://github.com/servo/servo/blob/master/tests/wpt/README.md#writing-new-tests.

The test should call setSelectionRange, but also that the `selectionStart` and `selectionEnd` values agree with other browsers, that is, something like:

```html
<!doctype html>
<textarea>

</textarea>
<script>
test(function() {
    let textarea = document.querySelector('textarea');
    assert_equals(textarea.selectionStart, 0);
    assert_equals(textarea.selectionEnd, 0);
    textarea.setSelectionRange(0, 1);
    assert_equals(textarea.selectionStart, 0);
    assert_equals(textarea.selectionEnd, 1);
}, "setSelectionRange on line boundaries");
</script>
```